### PR TITLE
add telemetry to track simulator auto install

### PR DESCRIPTION
**Pull request by adashen**

### CODE REVIEW

File: src/edge/simulator.ts [hunk 1]
```diff
@@ -56,10 +63,14 @@ export class Simulator {
...
 
     private static async validateSimulatorInstalled(outputChannel: vscode.OutputChannel = null): Promise<InstallReturn> {
+        const telemetryName = "simulatorInstalled";
         if (await Simulator.simulatorInstalled()) {
             return InstallReturn.Success;
         } else {
-            return await Simulator.installSimulatorWithPip(true, Constants.needSimulatorInstalledMsg, outputChannel);
+            TelemetryClient.sendEvent(`${telemetryName}.install`);
+            const installRes = await Simulator.installSimulatorWithPip(true, Constants.needSimulatorInstalledMsg, outputChannel);
+            TelemetryClient.sendEvent(`${telemetryName}.${InstallReturn[installRes]}`);
```
Comment (formulahendry) [inline src/edge/simulator.ts]: ${telemetryName}.install.${InstallReturn[installRes]} ?

Comment (adashen) [reply src/edge/simulator.ts]: telemetryName is already contains install. It only sent when install needed.

File: src/edge/simulator.ts [hunk 2]
```diff
@@ -38,16 +41,20 @@ export class Simulator {
             } else {
                 message = Constants.updateSimulatorMsg;
             }
+            type = "upgrade";
         } catch (error) {
             message =  Constants.needSimulatorInstalledMsg;
+            type = "install";
         }
 
         if (message) {
```
Comment (xinyiz1226) [inline src/edge/simulator.ts]: Need to check type value here
```suggestion
        if (message) {
```

Comment (adashen) [reply src/edge/simulator.ts]: type is the telemetry name. message has value means we need to install or update the iotedgehubdev tool. I will add default value to type as empty string to be safer.

File: src/edge/simulator.ts (hunk 1 above)
Comment (adashen) [reply src/edge/simulator.ts]: will update to be consistent with update scenario, Thx

Comment (formulahendry) [reply src/edge/simulator.ts]: So ``${telemetryName}.install`` could be ``${telemetryName}``?

Comment (adashen) [reply src/edge/simulator.ts]: updated.

File: src/edge/simulator.ts (hunk 2 above)
Comment (adashen) [reply src/edge/simulator.ts]: code updated

**Pull request merged**